src/OrleansLoggerProvider/LegacyOrleansLoggingFactoryExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using Microsoft.Extensions.Logging;
+using Orleans.Runtime;
+
+namespace Orleans.Extensions.Logging.Legacy
+{
+    public static class LegacyOrleansLoggingFactoryExtensions
+    {
+        /// <summary>
+        /// Add <see cref="LegacyOrleansLoggerProvider"> with event bulking feature to LoggerFactory
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="logConsumers">log consumers which user want to write log events to</param>
+        /// <param name="eventBulkingConfig">config for event bulking feature</param>
+        /// <returns></returns>
+        [Obsolete("Use Microsoft.Extensions.Logging built-in logger providers")]
+        public static ILoggingBuilder AddLegacyOrleansLogging(
```

Comment:
This is an ```ILoggingBuilder``` extension class and not ```ILoggingFactory```, please add ```ILoggingFactory``` implementation. SeriLog can be a good sample here perhaps: https://github.com/serilog/serilog-extensions-logging/tree/dev/src/Serilog.Extensions.Logging

Reply:
ILoggingBuilder is the 2.0 way of configuring logging. Serilog haven't migrated to it IMO because they haven't got to it. 

But take a look at Microsoft.Extensions.Logging.TraceSOurce code, they migrated to ILoggerBuilder. Let me @jdom here since he promoted that we migrated to ILoggerBuilder